ziti/identity.go

```diff
@@ -0,0 +1,105 @@
+/*
+	Copyright 2019 NetFoundry Inc.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+	https://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
+package ziti
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"github.com/michaelquigley/pfxlog"
+	"github.com/openziti/identity"
+	"sync"
+)
+
+var _ identity.Identity = &LazyIdentity{}
+
+// LazyIdentity will delay calling identity.LoadIdentity(config) till it is first accessed.
+type LazyIdentity struct {
```

Comment:
if we want this, i think it belongs in the identity project and not here

Reply:
Sure.